examples/advanced_agents/k8s_infra_monitor/src/nat_k8s_infra_monitor/event_collector_tool.py

```
@@ -0,0 +1,140 @@
+
+def _run_live(kubeconfig_path: str | None, event_limit: int) -> str:
+    """Execute kubectl commands to gather live cluster events."""
+    cmd_base = ["kubectl"]
+    if kubeconfig_path:
+        cmd_base.extend(["--kubeconfig", kubeconfig_path])
+
+    sections: list[str] = []
+
+    # Get warning events
+    try:
+        result = subprocess.run(
```

Ruff (0.15.6)

[error] `subprocess` call: check for execution of untrusted input

(S603)